src/Simplex/Messaging/Agent.hs

```diff
@@ -526,9 +531,14 @@ deleteConnectionsAsync_ onSuccess c connIds = case connIds of
 -- | Add connection to the new receive queue
 switchConnectionAsync' :: AgentMonad m => AgentClient -> ACorrId -> ConnId -> m ()
 switchConnectionAsync' c corrId connId =
-  withStore c (`getConn` connId) >>= \case
-    SomeConn _ DuplexConnection {} -> enqueueCommand c corrId connId Nothing $ AClientCommand $ APC SAEConn SWCH
-    _ -> throwError $ CMD PROHIBITED
+  withConnLock c connId "switchConnectionAsync" $
+    withStore c (`getConn` connId) >>= \case
+      SomeConn _ (DuplexConnection _ rqs@(rq :| _rqs) _) -> case findSwitchedRQ rqs of
+        Nothing -> do
+          void $ withStore' c $ \db -> setRcvQueueSwitchStatus db rq (Just RSSwitchStarted)
+          enqueueCommand c corrId connId Nothing $ AClientCommand $ APC SAEConn SWCH
+        Just _ -> throwError $ AGENT $ A_QUEUE "connection already switching"
```

Comment:
swap branches

Reply:
you should never change the order of predicates based on the contents, it's just confusing. Just should always come first, even if it's just throw.

Reply:
the reasoning that "what I do in Nothing" is the main behaviour, so let's put it first, is flawed. Think like you make a control flow diagram - you alway have consistent positioning of True/False branches there, it shouldn't depend on what's inside.